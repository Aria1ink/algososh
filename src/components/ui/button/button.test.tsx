import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe('Проверка элемента Button', function () {
  it('Без текста', function () {
    const element = renderer
      .create(<Button text=""/>)
      .toJSON();
    expect(element).toMatchSnapshot();
  });
  it('С текстом', function () {
    const element = renderer
      .create(<Button text="TEST"/>)
      .toJSON();
    expect(element).toMatchSnapshot();
  });
  it('С лоадером', function () {
    const element = renderer
      .create(<Button text="TEST" isLoader={true} />)
      .toJSON();
    expect(element).toMatchSnapshot();
  });
  it('Отключена', function () {
    const element = renderer
      .create(<Button text="TEST" disabled={true} />)
      .toJSON();
    expect(element).toMatchSnapshot();
  });
  it('Проверка колбека', function () {
    const testCallback = jest.fn();
    render(<Button text="TEST" onClick={testCallback} />);
    fireEvent.click(screen.getByText('TEST'));
    expect(testCallback).toHaveBeenCalled();
  });
});