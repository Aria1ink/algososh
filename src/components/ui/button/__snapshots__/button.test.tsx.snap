// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка состояний кнопки Без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Проверка состояний кнопки Отключена 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    TEST
  </p>
</button>
`;

exports[`Проверка состояний кнопки С лоадером 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Проверка состояний кнопки С текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    TEST
  </p>
</button>
`;

exports[`Проверка элемента Button Без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Проверка элемента Button Отключена 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    TEST
  </p>
</button>
`;

exports[`Проверка элемента Button С лоадером 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Проверка элемента Button С текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    TEST
  </p>
</button>
`;
